lib/improver/tests/utilities/test_warnings_handler.py

```diff
@@ -0,0 +1,140 @@
+    """ Dummy function to test """
+    if arg1 == 'User':
+        msg = 'Raising User error'
+        warnings.warn(msg)
+    return arg1, keyval1
+
+
+class Test__init__(IrisTest):
+
+    """Test __init__ of ManageWarnings."""
+
+    def test_basic(self):
+        """Test the default values are set correctly."""
+        plugin = ManageWarnings()
+        self.assertIsNone(plugin.messages)
+        self.assertIsNone(plugin.warning_types)
+        self.assertFalse(plugin.record)
+
+    def test_ignored_messages_not_list(self):
+        """Test Raises type error if ignored_messages is not a list."""
+        msg = "Expecting list of strings for ignored_messages"
+        with self.assertRaisesRegexp(TypeError, msg):
+            ManageWarnings(ignored_messages="Wrong")
+
+    def test_ignored_messages(self):
+        """Test set defaults set correctly when ignored_messages set."""
+        messages = ["Testing", "Testing2"]
+        plugin = ManageWarnings(ignored_messages=messages)
+        self.assertEqual(plugin.messages, messages)
+        self.assertEqual(len(plugin.warning_types), 2)
+        self.assertEqual(plugin.warning_types[0], UserWarning)
+        self.assertEqual(plugin.warning_types[1], UserWarning)
+
+    def test_ignored_messages_and_warning_types(self):
+        """Test the OK when ignored_messages and warning_types set."""
+        messages = ["Testing", "Testing2"]
+        warning_types = [UserWarning, PendingDeprecationWarning]
+        plugin = ManageWarnings(ignored_messages=messages,
+                                warning_types=warning_types)
+        self.assertEqual(plugin.messages, messages)
+        self.assertEqual(len(plugin.warning_types), 2)
+        self.assertEqual(plugin.warning_types[0], UserWarning)
+        self.assertEqual(plugin.warning_types[1], PendingDeprecationWarning)
+
+    def test_mismatch_warning_types(self):
+        """Test Raises error when warning_types does not match."""
+        messages = ["Testing", "Testing2"]
+        warning_types = [UserWarning]
+        msg = "Length of warning_types"
+        with self.assertRaisesRegexp(ValueError, msg):
+            ManageWarnings(ignored_messages=messages,
+                           warning_types=warning_types)
+
+    def test_record_is_true(self):
+        """Test record is set to True."""
+        plugin = ManageWarnings(record=True)
+        self.assertIsNone(plugin.messages)
+        self.assertIsNone(plugin.warning_types)
+        self.assertTrue(plugin.record)
+
+
+class Test__call__(IrisTest):
+
+    """Test __call__ of ManageWarnings."""
+
+    @ManageWarnings()
+    def test_basic(self):
+        """Test the Function still works with wrapper"""
+        argval, keyval = dummy_func('Test')
+        self.assertEqual(argval, 'Test')
+        self.assertIsNone(keyval)
+
+    @ManageWarnings(record=True)
+    def test_warning_list(self, warning_list=None):
+        """Test the Function still works with wrapper"""
```

Comment:
Need to update these docstrings, they are all the same at the moment.

Reply:
Opps, updated